astroplan/scheduling.py

```diff
@@ -140,6 +140,76 @@ def create_score_array(self, time_resolution=1*u.minute):
             score_array *= constraint(self.observer, targets, times)
         return score_array
 
+    def weighted_score_array(self, time_resolution=1*u.minute):
+        """
+        For each block, this returns a score for each time using the
+        formula (score*weight+score*weight+...)/(weight+weight+..)
+        """
+        # note: with this method, a non-boolean constraint with weight
+        # zero has the exact same effect if it was boolean
+        start = self.schedule.start_time
+        end = self.schedule.end_time
+        times = time_grid_from_range((start, end), time_resolution)
+        score_array = np.zeros((len(self.blocks), len(times)))
+        zeros = np.ones((len(self.blocks), len(times)), dtype=int)
```

Comment:
I should add some comments clarifying. If any constraint has a time where it has a value of zero, that gets stored in the zeros array, then the final scores get multiplied by the zeros array at the end. This preserves the intention that "if any constraint isn't satisfied the score becomes zero".